xbmc/GUIInfoManager.cpp

```diff
@@ -188,6 +188,18 @@ typedef struct
   int  val;
 } infomap;
 
+const infomap string_bools[] =   {{ "isempty",          STRING_IS_EMPTY },
+                                  { "equals",           STRING_EQUALS },
+                                  { "startswith",       STRING_STARTS_WITH },
+                                  { "endswith",         STRING_ENDS_WITH },
+                                  { "contains",         STRING_CONTAINS }};
+
+const infomap integer_bools[] =  {{ "equals",           INTEGER_EQUALS },
+                                  { "greater",          INTEGER_GREATER_THAN },
+                                  { "greaterorequals",  INTEGER_GREATER_OR_EQUALS },
+                                  { "less",             INTEGER_LESS_THAN },
+                                  { "lessorequals",     INTEGER_LESS_OR_EQUALS }};
```

Comment:
@phil65 IMO the methods are now very clear on what they are do. Before we use `substring` for substring comparison which for me indicates that it'll do a substring operation on the given string.